### PR TITLE
Minor updates to inventory input and error reporting

### PR DESCRIPTION
**Pull request by mpaiao**
Very minor updates to the inventory initialisation, that should have no impact on simulations.

### Description:

This fixes a couple of minor issues that shouldn't affect any simulations at this point.
1. Land use index should be an integer (currently not used, but it may be useful in the future).
2. Updated debugging reporting formats, so they are consistent with the variables being reported. 

### Checklist

*If this is your first time contributing, please read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/main/CONTRIBUTING.md) document.*

All checklist items must be checked to enable merging this pull request:

*Contributor*
- [ ] The in-code documentation has been updated with descriptive comments
- [ ] The documentation has been assessed to determine if updates are necessary

*Integrator*
- [ ] FATES PASS/FAIL regression tests were run
- [ ] Evaluation of test results for answer changes was performed and results provided

### Documentation

- [Technical Note](https://github.com/NGEET/fates-docs) update:
- [User's Guide](https://github.com/NGEET/fates-users-guide) update: 

### Test Results:

*CTSM (or) E3SM (specify which) test hash-tag:*

*CTSM (or) E3SM (specify which) baseline hash-tag:*

*FATES baseline hash-tag:*

*Test Output:*

### CODE REVIEW

File: main/FatesInventoryInitMod.F90 [hunk 1]
```diff
@@ -686,17 +686,18 @@ subroutine set_inventory_patch_type1(newpatch,pss_file_unit,ipa,ios,patch_name)
...
       real(r8)                                    :: p_age      ! Patch age [years]
       real(r8)                                    :: p_area     ! Patch area [fraction]
       integer                                     :: icwd       ! index for counting CWD pools
       integer                                     :: ipft       ! index for counting PFTs
       real(r8)                                    :: pftfrac    ! the inverse of the total number of PFTs
 
-      character(len=128),parameter    :: wr_fmt = &
-            '(F5.2,2X,A4,2X,F5.2,2X,F5.2,2X,F5.2,2X,F5.2,2X,F5.2,2X,F5.2,2X,F5.2,2X,F5.2,2X,F5.2,2X,F5.2,2X,F5.2)'
-
+      character(len=30),parameter    :: hd_fmt = &
```
Comment (rgknox) [inline main/FatesInventoryInitMod.F90:697]: You live dangerously @mpaiao , I think this was exactly 30 characters...

**Review by rgknox — APPROVED** [review]
These changes are straight forward and look good.  Thanks @mpaiao

File: main/FatesInventoryInitMod.F90 (hunk 1 above)
Comment (mpaiao) [reply main/FatesInventoryInitMod.F90:697]: I counted it ;)

Comment (ckoven) [reply main/FatesInventoryInitMod.F90:697]: @mpaiao Old school! I think our compilers let us use the `character(len=*), parameter :: var='xxx'` syntax, right?

Comment (mpaiao) [reply main/FatesInventoryInitMod.F90:697]: I think in the past I got error messages with using * and parameter, and somehow it stuck with me. I just tried a small test with gfortran-13 and it did not complain, should I update it?

Comment (rgknox) [reply main/FatesInventoryInitMod.F90:697]: its ok with me, as is

**Comment by glemieux** [comment]
Regression testing against `fates-sci.1.76.1_api.35.1.0-ctsm5.2.004` shows all expected tests passing b4b.  Note that this was tested in tandem with #1186.


results: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_pr1191-1186-fates`

**Pull request merged**